tensornetwork/tests/network_test.py

```diff
@@ -0,0 +1,673 @@
+  a = net.add_node(np.eye(2))
+  net.switch_backend(new_backend="tensorflow")
+  assert isinstance(a.tensor, tf.Tensor)
+
+
+def test_add_node_sanity_check(backend):
+  net = tensornetwork.TensorNetwork(backend=backend)
+  net.add_node(np.eye(2), "a")
+  net.check_correct()
+
+
+def test_add_node_in_network(backend):
+  net = tensornetwork.TensorNetwork(backend=backend)
+  a = net.add_node(np.eye(2))
+  assert a in net
+
+
+def test_add_node_twice_raise_value_error(backend):
+  net = tensornetwork.TensorNetwork(backend=backend)
+  a = net.add_node(tensornetwork.CopyNode(3, 3))
+  with pytest.raises(ValueError):
+    net.add_node(a)
+
+
+def test_add_copy_node(backend):
+  net = tensornetwork.TensorNetwork(backend=backend)
+  a = net.add_node(
+      tensornetwork.CopyNode(3, 3, name="TestName", axis_names=['a', 'b', 'c']))
+  assert a in net
+
+
+def test_connect_dangling_edges(backend):
+  net = tensornetwork.TensorNetwork(backend=backend)
+  a = net.add_node(np.ones((3,)))
+  b = net.add_node(np.ones((3,)))
+  edge = net.connect(a[0], b[0])
+  net.check_correct()
+  assert a.edges == b.edges
+  assert edge in net
+
+
+def test_connect_axis_names(backend):
+  net = tensornetwork.TensorNetwork(backend=backend)
+  a = net.add_node(np.ones((3,)), name="a", axis_names=["one"])
+  b = net.add_node(np.ones((3,)), name="b", axis_names=["one"])
+  edge = net.connect(a["one"], b["one"])
+  net.check_correct()
+  assert a.edges == b.edges
+  assert edge in net
+
+
+def test_connect_twice_edge_axis_value_error(backend):
+  net = tensornetwork.TensorNetwork(backend=backend)
+  a = net.add_node(np.array([2.]), name="a")
+  b = net.add_node(np.array([2.]), name="b")
+  net.connect(a[0], b[0])
+  with pytest.raises(ValueError):
+    net.connect(a[0], b[0])
+
+
+def test_connect_same_edge_value_error(backend):
+  net = tensornetwork.TensorNetwork(backend=backend)
+  a = net.add_node(np.eye(2))
+  with pytest.raises(ValueError):
+    net.connect(a[0], a[0])
+
+
+def test_disconnect_edge(backend):
+  net = tensornetwork.TensorNetwork(backend=backend)
+  a = net.add_node(np.array([1.0] * 5), "a")
+  b = net.add_node(np.array([1.0] * 5), "b")
+  e = net.connect(a[0], b[0])
+  assert not e.is_dangling()
+  dangling_edge_1, dangling_edge_2 = net.disconnect(e)
+  net.check_correct(check_connected=False)
+  assert dangling_edge_1.is_dangling()
+  assert dangling_edge_2.is_dangling()
+  assert a.get_edge(0) == dangling_edge_1
+  assert b.get_edge(0) == dangling_edge_2
+
+
+def test_disconnect_edge_not_in_network(backend):
+  net = tensornetwork.TensorNetwork(backend=backend)
+  a = net.add_node(np.eye(2))
+  b = net.add_node(np.eye(2))
+  edge = net.connect(a[0], b[0])
+  net.disconnect(edge)
+  assert edge not in net
+
+
+def test_disconnect_dangling_edge_value_error(backend):
+  net = tensornetwork.TensorNetwork(backend=backend)
+  a = net.add_node(np.eye(2))
+  with pytest.raises(ValueError):
+    net.disconnect(a[0])
+
+
+def test_remove_trace_edge():
+  #todo
```

Comment:
Why are these todos?

Reply:
@Thenerdstation  > * `network_test.py`: contains unit tests for all methods of `TensorNetwork` class. Some to-do's are added for missing tests, for which - if agreed by reviewers - I can later make an Issue.

Reply:
Somehow my quoting of my previous message got messed up. @Thenerdstation, by splitting up the tests, I noticed some methods of TensorNetwork do not have the most basic of unit tests. Some of them are private, but I'm a bit of an annoying test guy and would like to see basic unittests for all of them :). My plan was to make an issue for this after this PR and leave it as a good first issue. If the todos bother you I can remove them.